cmd/natc/natc.go

```diff
@@ -281,35 +358,39 @@ func (c *connector) generateDNSResponse(req *dnsmessage.Message, from tailcfg.No
 			}
 			found = true
 			if want6 {
-				err = b.AAAAResource(
+				if err := b.AAAAResource(
 					dnsmessage.ResourceHeader{Name: q.Name, Class: q.Class, TTL: 5},
 					dnsmessage.AAAAResource{AAAA: ip.As16()},
-				)
+				); err != nil {
+					return nil, err
+				}
 			} else {
-				err = b.AResource(
+				if err := b.AResource(
 					dnsmessage.ResourceHeader{Name: q.Name, Class: q.Class, TTL: 5},
 					dnsmessage.AResource{A: ip.As4()},
-				)
+				); err != nil {
+					return nil, err
+				}
 			}
-			break
 		}
 		if !found {
-			err = errors.New("no address found")
+			return nil, errors.New("no address found")
 		}
```

Comment:
Looks like you can just return normal success if `!found`. `NXDOMAIN` should be handled somewhere if an actual resolver error happens.